test/crdsvalidation/kongpluginbindings/kongpluginbindings_test.go

```diff
@@ -29,11 +29,13 @@ func TestKongPluginBindings(t *testing.T) {
 				t.Run(tc.Name, func(t *testing.T) {
 					cl := cl.KongPluginBindings(tc.KongPluginBinding.Namespace)
 					kpb, err := cl.Create(ctx, &tc.KongPluginBinding, metav1.CreateOptions{})
-					if tc.ExpectedErrorMessage == nil {
-						assert.NoError(t, err)
+					if err != nil {
```

Comment:
If we failed to create the resource then we don't need to schedule the deletion, no?

Can you elaborate why we need to change this? In case there was an error (in the previous version) line 33 will assert and log and then we'll schedule a testing cleanup which will ignore the not found error. Am I missing something?

Reply:
With the previous logic, if `expectedErrorMessage` != nil (error expected), but the creation succeeded (no error occurred), we perform no cleanup. This way we always perform the cleanup, no matter the expected outcome. Does it make sense?

Reply:
Oh I see. Yup, that makes sense. 👍